problem_builder/instructor_tool.py

```diff
@@ -105,9 +134,95 @@ def student_view(self, context=None):
             _('Rating Question'): 'RatingBlock',
             _('Long Answer'): 'AnswerBlock',
         }
+        block_types = ('pb-mcq', 'pb-rating', 'pb-answer')
+        flat_block_tree = []
+
+        def get_block_id(block):
+            """
+            Return ID of `block`, taking into account needs of both LMS/CMS and workbench runtimes.
+            """
+            usage_id = block.scope_ids.usage_id
+            # Try accessing block ID. If usage_id does not have it, return usage_id itself
+            return unicode(getattr(usage_id, 'block_id', usage_id))
+
+        def get_block_name(block):
+            """
+            Return name of `block`.
+
+            Try attributes in the following order:
+              - block.question
+              - block.name (fallback for old courses)
+              - block.display_name
+              - block ID
+            """
+            # - Try "question" attribute:
+            block_name = getattr(block, 'question', None)
+            if not block_name:
+                # Try question ID (name):
+                block_name = getattr(block, 'name', None)
+            if not block_name:
+                # - Try display_name:
+                block_name = getattr(block, 'display_name', None)
+            if not block_name:
+                # - Default to ID:
+                block_name = get_block_id(block)
+            return block_name
```

Comment:
@itsjeyd nit: could be more concise and DRY:

    for attribute in ('question', 'name', 'display_name'):
        if getattr('block', attribute, None):
            return getattr('block', attribute, None)

    return get_block_id(block)

Not insisting on this though.